controllers/component_controller_test.go

```diff
@@ -2093,6 +2094,325 @@ var _ = Describe("Component controller", func() {
 		})
 	})
 
+	Context("Create Private Component with basic field set", func() {
+		It("Should create successfully and update the Application", func() {
+			ctx := context.Background()
+
+			applicationName := HASAppName + "24"
+			componentName := HASCompName + "24"
+
+			// Create a git secret
+			tokenSecret := &corev1.Secret{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "Secret",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      componentName,
+					Namespace: HASAppNamespace,
+				},
+				StringData: map[string]string{
+					"password": "valid-token", // token tied to mock implementation in devfile/library
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, tokenSecret)).Should(Succeed())
+
+			createAndFetchSimpleApp(applicationName, HASAppNamespace, DisplayName, Description)
+
+			hasComp := &appstudiov1alpha1.Component{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "appstudio.redhat.com/v1alpha1",
+					Kind:       "Component",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      componentName,
+					Namespace: HASAppNamespace,
+				},
+				Spec: appstudiov1alpha1.ComponentSpec{
+					ComponentName: ComponentName,
+					Application:   applicationName,
+					Secret:        componentName,
+					Source: appstudiov1alpha1.ComponentSource{
+						ComponentSourceUnion: appstudiov1alpha1.ComponentSourceUnion{
+							GitSource: &appstudiov1alpha1.GitSource{
+								URL: SampleRepoLink,
```

Comment:
the test says for private repo, is this `samplerepolink` for public repo testing tho?

Reply:
Actually it doesn't matter what repo we provide here, we are testing the token "valid-token" which returns a mock devfile from  See https://github.com/devfile/library/blob/main/pkg/util/mock.go#L250. But I changed this line to a private repo to avoid confusion and added a comment to the mock file. 👍